### PR TITLE
docs: Add recipe for Using CSS Modules in Styling section

### LINKED ISSUE #14896: [docs][recipes] Styling: Using CSS modules
## Summary

To improve key learning workflows in the Recipes section of the docs, the "Styling" recipe should be broken up into smaller, more actionable parts. This issue is about the item **Using CSS modules**, coming after the Emotion recipe requested in https://github.com/gatsbyjs/gatsby/issues/14895. 

The current recipe has this text (it will be replaced by Styling recipes inline): 

> There are so many ways to add styles to your website; Gatsby supports almost every possible option, through official and community plugins.
> - Walk through adding global styles to an example site in tutorial part two
>    - More on global styles with standard CSS files
>    - More on global styles with CSS-in-JS
>    - More on global styles with CSS files and no layout component
> - Use the CSS-in-JS library Glamor
> - Use the CSS-in-JS library Styled Components
> - Use CSS Modules

The actionable version of this CSS module recipe should show how to work with them _inline_ as concisely as possible. It should include code examples showing what makes a CSS module and how to import one in a component. The CSS modules docs page can be used as a reference: https://www.gatsbyjs.org/docs/css-modules/

Recipes now follow this format:

- Requirements or prerequisites 
- Step-by-step instructions
- Relevant links to any relevant information
    - More on using CSS Modules https://www.gatsbyjs.org/docs/css-modules/
    - Link to a live example 

This recipe should go into the "Styling" section of [`docs/docs/recipes.md`](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/recipes.md), replacing the existing text. There is a template in the contributing docs: https://www.gatsbyjs.org/contributing/docs-templates/#recipes

For reference, the new recipe collection is also listed in a spreadsheet: https://docs.google.com/spreadsheets/d/1yNUQi_-3TsuDtE98hCIJQvLZ6X1YHlDmC-EJwOGxiHk/edit#gid=1252007048

### PR DESCRIPTION
**Pull request by mehtavishwa30**
## Description

This is a short actionable recipe for the `Using CSS modules` item in the `Styling` section of in `Recipes`

## Related Issues

Fixes #14896

### CODE REVIEW

File: docs/docs/recipes.md [hunk 1]
```diff
@@ -343,6 +343,51 @@ export default () => (
 - [More on Using Styled Components](/docs/styled-components/)
 - [Egghead lesson](https://egghead.io/lessons/gatsby-style-gatsby-sites-with-styled-components)
 
+### Using CSS Modules
+
+#### Prerequisites
+
+- A [Gatsby site](/docs/quick-start/) with an index page component
+- An existing
```
Comment (wardpeet) [inline docs/docs/recipes.md]: I think you forgot to add a few more words here 😄

File: docs/docs/recipes.md [hunk 2]
```diff
@@ -343,6 +343,51 @@ export default () => (
 - [More on Using Styled Components](/docs/styled-components/)
 - [Egghead lesson](https://egghead.io/lessons/gatsby-style-gatsby-sites-with-styled-components)
 
+### Using CSS Modules
+
+#### Prerequisites
+
+- A [Gatsby site](/docs/quick-start/) with an index page component
```
Comment (sidharthachatterjee) [inline docs/docs/recipes.md]:
```suggestion
- An existing [Gatsby site](/docs/quick-start/) with an index page component
```

File: docs/docs/recipes.md (hunk 1 above)
Comment (sidharthachatterjee) [inline docs/docs/recipes.md]:
```suggestion

```

Comment (mehtavishwa30) [reply docs/docs/recipes.md]: Sorry for not getting to this earlier. Looks like I missed something here.

**Comment by mehtavishwa30** [comment]
Thanks for the review @wardpeet! Also, @sidharthachatterjee appreciate you clearing up the prerequisite suggestion in my stead. 
🎉

File: docs/docs/recipes.md (hunk 1 above)
Comment (wardpeet) [reply docs/docs/recipes.md]: No need to say sorry 😉  We thank you for the work you've done! Our users will be super happy about this! We appreciate the work! 🥇

**Pull request merged**